### PR TITLE
ui: Cache avatar image to improve performance

### PR DESCRIPTION
**Pull request by connected-adietz**
- Cache the user's image upon login and app startup
- Show default profile image when user image is not reachable
- Resolves stall in app when not connected to internet and image is attempted to be fetched

### CODE REVIEW

**Comment by mhuang-connectedqa** [comment]
<h3>Errors: 0, Warnings: 6</h3>
    Details: <strong><a href='https://circleci.com/gh/mozilla-services/guardian-vpn-windows/392/artifacts/0/Users/circleci/project/test/result/Output.log' target='_blank'>details</a></strong>

File: ui/src/Manager.cs [hunk 1]
```diff
@@ -233,5 +245,57 @@ public static void InitializeSettings()
...
+                            try
+                            {
+                                using (HttpWebResponse response = (HttpWebResponse)req.GetResponse())
+                                {
+                                    image = new BitmapImage(new Uri(Account.Config.FxALogin.User.Avatar));
+                                }
+                            }
+                            catch (WebException)
+                            {
+                                image = new BitmapImage(new Uri(Path.Combine(Environment.CurrentDirectory, @"..\..\..\src\UI\Resources\Icons\Generic\default.png")));
```
Comment (connected-nkozul) [inline ui/src/Manager.cs]: Could you rename `default.png` to something more descriptive, like `default-avatar.png`?

File: ui/src/UI/Components/Card/Avatar/Avatar.xaml.cs [hunk 2]
```diff
@@ -76,15 +84,46 @@ public double Size
...
+                            using (HttpWebResponse response = (HttpWebResponse)req.GetResponse())
+                            {
+                                image = new BitmapImage(new Uri(Manager.Account.Config.FxALogin.User.Avatar));
+                            }
+
+                            Manager.Cache.Set("avatarImage", image, policy);
+                            return (BitmapImage)image;
+                        }
+                        catch (Exception)
+                        {
```
Comment (connected-nkozul) [inline ui/src/UI/Components/Card/Avatar/Avatar.xaml.cs]: `ErrorHandling.ErrorHandler.Handle(e, ErrorHandling.LogLevel.Debug);`

File: ui/src/UI/Components/Card/Avatar/Avatar.xaml.cs [hunk 3]
```diff
@@ -76,15 +84,46 @@ public double Size
...
+                            using (HttpWebResponse response = (HttpWebResponse)req.GetResponse())
+                            {
+                                image = new BitmapImage(new Uri(Manager.Account.Config.FxALogin.User.Avatar));
+                            }
+
+                            Manager.Cache.Set("avatarImage", image, policy);
+                            return (BitmapImage)image;
+                        }
+                        catch (Exception)
```
Comment (connected-nkozul) [inline ui/src/UI/Components/Card/Avatar/Avatar.xaml.cs]: `catch (Exception e)`

File: ui/src/Manager.cs [hunk 4]
```diff
@@ -233,5 +245,57 @@ public static void InitializeSettings()
...
+                        }
+
+                        Cache.Set("avatarImage", image, policy);
+                    });
+                });
+            }
+        }
+
+        /// <summary>
+        /// Initializes cache for avatar image.
```
Comment (connected-nkozul) [inline ui/src/Manager.cs]: Initializes, or clears?

File: ui/src/Manager.cs [hunk 5]
```diff
@@ -233,5 +245,57 @@ public static void InitializeSettings()
         {
             Settings = new Settings(ProductConstants.SettingsFile);
         }
+
+        /// <summary>
+        /// Initializes cache for avatar image.
+        /// </summary>
+        public static void InitializeCache()
```
Comment (connected-nkozul) [inline ui/src/Manager.cs]: If you believe there should be a need for more caching later, remove "for avatar image" from function description.

If not, rename `InitializeCache()` to `InitializeAvatarCache()` or whatever you think is best.

File: ui/src/FxA/Account.cs [hunk 6]
```diff
@@ -118,6 +121,7 @@ public void Logout(bool removeDevice = false)
                     var devices = new FxA.Devices();
                     devices.RemoveDevice(Manager.Account.Config.FxALogin.PublicKey, silent: true);
                 }
+
```
Comment (connected-nkozul) [inline ui/src/FxA/Account.cs]: nit: Remove new line

**Comment by mhuang-connectedqa** [comment]
<h3>Errors: 0, Warnings: 0</h3>
    Details: <strong><a href='https://circleci.com/gh/mozilla-services/guardian-vpn-windows/401/artifacts/0/Users/circleci/project/test/result/Output.log' target='_blank'>details</a></strong>

**Comment by mhuang-connectedqa** [comment]
<h3>Errors: 0, Warnings: 0</h3>
    Details: <strong><a href='https://circleci.com/gh/mozilla-services/guardian-vpn-windows/402/artifacts/0/Users/circleci/project/test/result/Output.log' target='_blank'>details</a></strong>

**Comment by mhuang-connectedqa** [comment]
<h3>Errors: 0, Warnings: 0</h3>
    Details: <strong><a href='https://circleci.com/gh/mozilla-services/guardian-vpn-windows/403/artifacts/0/Users/circleci/project/test/result/Output.log' target='_blank'>details</a></strong>

File: ui/src/UI/Components/Card/Avatar/Avatar.xaml.cs [hunk 7]
```diff
@@ -76,15 +84,47 @@ public double Size
...
+                        }
+                        catch (Exception e)
+                        {
+                            ErrorHandling.ErrorHandler.Handle(e, ErrorHandling.LogLevel.Debug);
+                        }
+                    }
                 }
             }
+
+            return new BitmapImage(new Uri(System.IO.Path.Combine(Environment.CurrentDirectory, @"..\..\..\src\UI\Resources\Icons\Generic\default-avatar.png")));
```
Comment (connected-mhuang) [inline ui/src/UI/Components/Card/Avatar/Avatar.xaml.cs]: I think these lines of code are kind of similar to those in Manager.cs file. Could you create a method to get the image through avatar url? This method can be used in many places and can be easily to maintain.

**Comment by mhuang-connectedqa** [comment]
<h3>Errors: 0, Warnings: 0</h3>
    Details: <strong><a href='https://circleci.com/gh/mozilla-services/guardian-vpn-windows/405/artifacts/0/Users/circleci/project/test/result/Output.log' target='_blank'>details</a></strong>

**Pull request merged**